### PR TITLE
MM-11605: Do not show archived channels in the "more..." public channels selector

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Do not show archived channels in the "more..." public channels selector

#### Ticket Link
[MM-11605](https://mattermost.atlassian.net/browse/MM-11605)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed

### CODE REVIEW

File: components/more_channels/more_channels.jsx [hunk 1]
```diff
@@ -172,6 +172,13 @@ export default class MoreChannels extends React.Component {
             </TeamPermissionGate>
         );
 
+        let channelsList = [];
+        if (search) {
+            channelsList = searchedChannels.filter((c) => c.delete_at === 0);
+        } else {
+            channelsList = channels.filter((c) => c.delete_at === 0);
+        }
```
Comment (saturninoabril) [inline components/more_channels/more_channels.jsx]: Should the filtering be done outside the render function?  Like,
-  on `searchedChannels` [before setting the state](https://github.com/mattermost/mattermost-webapp/blob/6c348ef6e830490698a616a4ccb3aca44067a7ef/components/more_channels/more_channels.jsx#L114).
- on `channels` [before passing as props](https://github.com/mattermost/mattermost-webapp/blob/v5.2.0-rc2/components/more_channels/index.js#L17)

Comment (jespino) [reply components/more_channels/more_channels.jsx]: Good point

Comment (jespino) [reply components/more_channels/more_channels.jsx]: Done! PTAL

File: components/more_channels/more_channels.jsx [hunk 2]
```diff
@@ -121,6 +121,10 @@ export default class MoreChannels extends React.Component {
         this.searchTimeoutId = searchTimeoutId;
     }
 
+    setSearchResults = (channels) => {
```
Comment (lieut-data) [inline components/more_channels/more_channels.jsx]: Out of scope for this PR, but it's too bad we can't filter this out server-side to improve performance.

**Pull request merged**